### PR TITLE
feat(TDP-5982): remove ExportStrategy interface

### PR DESCRIPTION
**Pull request by hmdebenque**
**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [ ] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

### CODE REVIEW

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/SampleExportStrategy.java [hunk 1]
```diff
@@ -11,10 +11,21 @@
...
-public interface SampleExportStrategy extends ExportStrategy {
+public interface SampleExportStrategy extends Predicate<ExportParameters> {
+
+    /**
+     * Execute export strategy with given parameters. Callers are expected to ensure {@link #test(Object)}
+     * returns <code>true</code> before calling this method.
+     * @param parameters Export parameters (can <b>not</b> be null).
+     * @return A {@link StreamingResponseBody} that streams the export data to the provided output stream.
+     */
+    StreamingResponseBody execute(ExportParameters parameters);
```
Comment (aleboulanger) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/SampleExportStrategy.java]: You can enforce null check by adding `@NotNull` annotation in order to show in the IDE that the parameters could not be null when this method is used.

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/BaseTransformationService.java [hunk 2]
```diff
@@ -99,10 +100,10 @@ StreamingResponseBody executeSampleExportStrategy(final ExportParameters paramet
         }
     }
 
-    protected <T extends ExportStrategy> Optional<T> chooseExportStrategy(OrderedBeans<T> exportStrategies,
+    protected <T extends Predicate<ExportParameters>> Optional<T> chooseExportStrategy(OrderedBeans<T> exportStrategies,
             final ExportParameters parameters) {
         return exportStrategies //
-                .filter(exportStrategy -> exportStrategy.accept(parameters)) //
```
Comment (aleboulanger) [inline dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/BaseTransformationService.java]: why you want to change the signature of this method ?

File: dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/SampleExportStrategy.java (hunk 1 above)
Comment (hmdebenque) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/SampleExportStrategy.java]: I don't like adding IDE oriented tools to code. Moreover @NotNull is not standardized and only exists in data validation libraries (javax.validation in data-prep).

Comment (hmdebenque) [reply dataprep-transformation/src/main/java/org/talend/dataprep/transformation/service/export/SampleExportStrategy.java]: We also do not need to have defensive programming inside different parts of dataprep

**Pull request merged**